src/services/push-service.ts

```diff
@@ -190,11 +190,31 @@ export const buildAssetToDeployTask = async (
 };
 
 export const prepareEnvironmentTask = async (ctx: PushCommandTasksContext) => {
-  const signedCloudStorageUrl = await getSignedStorageUrl(ctx.appVersionId);
-  const archiveContent = readFileData(ctx.archivePath!);
-  ctx.signedCloudStorageUrl = signedCloudStorageUrl;
-  ctx.archiveContent = archiveContent;
-  ctx.showUploadAssetTask = true;
+  try {
+    const signedCloudStorageUrl = await getSignedStorageUrl(ctx.appVersionId);
+    const archiveContent = readFileData(ctx.archivePath!);
+    ctx.signedCloudStorageUrl = signedCloudStorageUrl;
+    ctx.archiveContent = archiveContent;
+    ctx.showUploadAssetTask = true;
+  } catch (error: any | HttpError) {
+    if (error instanceof HttpError && error.code === 409) {
+      ctx.stopDeploymentTaskOnConflictError = true;
+      return;
+    }
+
+    throw error;
+  }
+};
+
+export const handleDenyConflictingDeploymentTask = (
```

Comment:
Name is misleading here, there is no task to be done. This just marks a failure due to a conflict, right?

Reply:
agree. changed to `stopNewDeploymentWhileExistingInProgress`